src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/AccountController.cs

```diff
@@ -279,6 +279,11 @@ public async Task<IActionResult> ConfirmEmail(string userId, string code)
             }
 
             var result = await this._accountService.ConfirmEmailAsync(user, code).ConfigureAwait(false);
+            if (!result.Succeeded)
+            {
+                throw new InvalidOperationException();
+            }
+
             return this.View(result.Succeeded ? "ConfirmEmail" : "Error");
```

Comment:
you just checked `result.Succeeded` in line 282. So here all you need to do is returning View. So just adding the code, that I wrote before :)